src/tests/ftest/cart/dual_iface_server.c

```diff
@@ -0,0 +1,434 @@
+/*
+ * (C) Copyright 2018-2021 Intel Corporation.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause-Patent
+ */
+/**
+ * Server utilizing dual interfaces to check connectivity
+ */
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <unistd.h>
+#include <assert.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <semaphore.h>
+#include <cart/api.h>
+#include <cart/types.h>
+
+#include "tests_common.h"
+
+#define MY_BASE 0x010000000
+#define MY_VER  0
+
+#define NUM_SERVER_CTX 8
+
+#define RPC_DECLARE(name)						\
+	CRT_RPC_DECLARE(name, CRT_ISEQ_##name, CRT_OSEQ_##name)		\
+	CRT_RPC_DEFINE(name, CRT_ISEQ_##name, CRT_OSEQ_##name)
+
+enum {
+	RPC_PING = CRT_PROTO_OPC(MY_BASE, MY_VER, 0),
+	RPC_SHUTDOWN
+} rpc_id_t;
+
+#define CRT_ISEQ_RPC_PING	/* input fields */		 \
+	((uint64_t)		(src_rank)		CRT_VAR) \
+	((uint64_t)		(dst_tag)		CRT_VAR)
+
+#define CRT_OSEQ_RPC_PING	/* output fields */		 \
+	((uint64_t)		(rc)			CRT_VAR)
```

Comment:
This is unsigned, you may be returning negative values.  Do you want/need sign extension?

Reply:
will change to signed